src/main/java/backend/UpdateController.java

```diff
@@ -61,6 +62,15 @@ public void processAndRefresh(List<FilterPanel> filterPanels) {
                     if (toUpdate.isEmpty()) {
                         // If no issues requiring metadata update, just run the filter and sort.
                         logic.updateUI(processFilters(filterExprs));
+                        ArrayList<CompletableFuture<Boolean>> stub = new ArrayList<>();
+                        Futures.sequence(stub)
+                                .thenRun(() -> {
+                                    if (isResumeTimer) {
+                                        logic.refreshTimer.resume();
+                                    }
+                                })
+                                .thenCompose(n -> logic.getRateLimitResetTime())
+                                .thenApply(logic::updateRateLimits);
```

Comment:
@ndt93 Yes, i agree. I think it is necessary to have this. Consider if the case where the refresh timer doesn't stop. And the refreshing going to take more than 1mins, the refresh timer will trigger another refresh before the first one is completed. I think this is kind of tricky, I will revert it to the old implementation for now. Will discuss in detail in future.